lib/OpenQA/Worker/Cache.pm

```diff
@@ -350,14 +350,20 @@ sub check_limits {
         $sql    = "SELECT size, filename FROM assets WHERE downloading = 0 ORDER BY last_use asc";
         $sth    = $dbh->prepare($sql);
         $result = $dbh->selectrow_hashref($sql);
-
-        foreach my $asset ($result) {
-            if (purge_asset($asset->{filename})) {
-                $cache_real_size -= $asset->{size};
-                log_debug "Reclaiming " . $asset->{size} . " from $cache_real_size to make space for $limit";
-            }    # purge asset will die anyway in case of failure.
-            last if ($cache_real_size < $limit);
+        if ($result) {
+            foreach my $asset ($result) {
```

Comment:
This loop appears to do nothing, what am i missing? Is aliasing `$result` to `$asset` somehow significant here?